cranelift/codegen/src/isa/aarch64/inst/mod.rs

```diff
@@ -1589,92 +1655,81 @@ impl Inst {
+                    AtomicRMWLoopOp::Umax => "umax",
+                    AtomicRMWLoopOp::Xchg => "xchg",
                 };
-
-                if op_str.is_empty() {
-                    match op {
-                        AtomicRMWLoopOp::Xchg => r_dst = r_arg2,
-                        AtomicRMWLoopOp::Nand => {
-                            loop_str.push_str(&format!("and {}, {}, {}; ", r_dst, r_tmp, r_arg2));
-                            loop_str.push_str(&format!("mvn {}, {}; ", r_dst, r_dst));
-                        }
-                        _ => {
-                            if (op == AtomicRMWLoopOp::Smin || op == AtomicRMWLoopOp::Smax)
-                                && (ty == I8 || ty == I16)
-                            {
-                                loop_str
-                                    .push_str(&format!("sxt{} {}, {}; ", ty_suffix, r_tmp, r_tmp));
-                                loop_str.push_str(&format!(
-                                    "cmp {}, {}, sxt{}; ",
-                                    r_tmp, r_arg2, ty_suffix
-                                ));
-                            } else {
-                                loop_str.push_str(&format!("cmp {}, {}; ", r_tmp, r_arg2));
-                            }
-                            let cond = match op {
-                                AtomicRMWLoopOp::Smin => "lt",
-                                AtomicRMWLoopOp::Smax => "gt",
-                                AtomicRMWLoopOp::Umin => "lo",
-                                AtomicRMWLoopOp::Umax => "hi",
-                                _ => unreachable!(),
-                            };
-                            loop_str.push_str(&format!(
-                                "csel {}, {}, {}, {}; ",
-                                r_dst, r_tmp, r_arg2, cond
-                            ));
-                        }
-                    };
-                } else {
-                    loop_str.push_str(&format!("{} {}, {}, {}; ", op_str, r_dst, r_tmp, r_arg2));
-                }
-                loop_str.push_str(&format!(
-                    "stlxr{} {}, {}, [{}]; ",
-                    ty_suffix, r_status, r_dst, r_addr
-                ));
-                loop_str.push_str(&format!("cbnz {}, 1b", r_status));
-                loop_str
-            }
-            &Inst::AtomicCAS { rs, rt, rn, ty } => {
+                let addr = pretty_print_ireg(addr, OperandSize::Size64, allocs);
+                let operand = pretty_print_ireg(operand, OperandSize::Size64, allocs);
+                let oldval = pretty_print_ireg(oldval.to_reg(), OperandSize::Size64, allocs);
+                let scratch1 = pretty_print_ireg(scratch1.to_reg(), OperandSize::Size64, allocs);
+                let scratch2 = pretty_print_ireg(scratch2.to_reg(), OperandSize::Size64, allocs);
+                format!(
+                    "atomic_rmw_loop_{}_{} addr={} operand={} oldval={} scratch1={} scratch2={}",
```

Comment:
BTW while this is unfortunately a step back (in the sense that the previous implementation prints the actual assembly code that is generated), since the operation is quite complex, I think that this simplification is acceptable.

Reply:
Indeed, I was having trouble keeping it up-to-date as I was adding in the registers and chose instead to simplify it here; it felt odd to have a separate representation that wasn't checked against the actual emitted code.